File: typescript/kata/two-sum.ts
function twoSum(nums: number[], target: number): number[] {
  const seen: Record<number, number> = {};

  for (let i = 0; i < nums.length; i++) {
    const diff = target - nums[i];

    if (seen[diff] !== undefined) {
      return [seen[diff], i];
    }

    seen[nums[i]] = i;
  }

  return [];
}

const nums = [2, 7, 11, 15];
const target = 9;

console.log(twoSum(nums, target));

File: java/kata/src/main/java/br/com/ari/algo/algorithms/TwoSum.java
package br.com.ari.algo.algorithms;

import java.util.HashMap;
import java.util.Map;

public class TwoSum {

    private TwoSum() {}

    public static int[] solve(int[] nums, int target) {
        Map<Integer, Integer> seen = new HashMap<>();

        for (int i = 0; i < nums.length; i++) {
            int diff = target - nums[i];

            if (seen.containsKey(diff)) {
                return new int[]{ seen.get(diff), i };
            }

            seen.put(nums[i], i);
        }

        return new int[2];
    }

}
